### PR TITLE
Move server code to ai-playground

### PR DESCRIPTION
**Pull request by yansavitski**
- [x] ai_playground server code to be moved to package
- [x] move `streamFactory` in `@kbn/ml-response-stream/server` 
- [X] investigate whether we can use it - we can, cause the package doesn't have any dependencies
- [x] ai-assist server code to be moved to package

### CODE REVIEW

File: packages/kbn-ai-playground/server/routes.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import { fetchFields } from '../lib/fetch_query_source_fields';
+import Assist, { ConversationalChain, Prompt } from '@elastic/ai-assist';
+import { ChatOpenAI } from '@elastic/ai-assist/models';
+import { streamFactory } from '@kbn/ml-response-stream/server';
+import { Logger } from '@kbn/logging';
+import { IRouter, RequestHandler } from '@kbn/core/server';
+
+export function registerAIPlaygroundRoutes(
+  { log, router }: { log: Logger; router: IRouter },
+  paths: { querySourceFields: string; chat: string; apiKey: string },
```
Comment (joemcelroy) [inline packages/kbn-ai-playground/server/routes.ts]: nitpick: i think you should just do a base url here instead

File: packages/kbn-ai-playground/server/routes.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { fetchFields } from '../lib/fetch_query_source_fields';
+import Assist, { ConversationalChain, Prompt } from '@elastic/ai-assist';
+import { ChatOpenAI } from '@elastic/ai-assist/models';
+import { streamFactory } from '@kbn/ml-response-stream/server';
```
Comment (Samiul-TheSoccerFan) [inline packages/kbn-ai-playground/server/routes.ts:11]: Are we still planning to export `streamFactory` our of `@kbn/ml-response-stream/server` or use it as it is?

File: packages/kbn-ai-playground/server/routes.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { fetchFields } from '../lib/fetch_query_source_fields';
+import Assist, { ConversationalChain, Prompt } from '@elastic/ai-assist';
```
Comment (Samiul-TheSoccerFan) [inline packages/kbn-ai-playground/server/routes.ts]: I thought the idea is to take out server codes from `elastic/ai-assist` and move it into our own server codes so we do not have any more dependency on `elastic/ai-assist`. Please correct me if I am wrong.

Comment (yansavitski) [reply packages/kbn-ai-playground/server/routes.ts]: Yeah, I got it wrong)

File: packages/kbn-ai-playground/server/routes.ts (hunk 2 above)
Comment (yansavitski) [reply packages/kbn-ai-playground/server/routes.ts:11]: I thought the plan was to check dependencies and if it has then will think about next steps, but as it doesn't we can just use this function

File: packages/kbn-ai-playground/utils/api.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
```
Comment (joemcelroy) [inline packages/kbn-ai-playground/utils/api.ts]: nitpick - can we move the the server only parts into a server/utils here? just so its easier to navigate.

Comment (yansavitski) [reply packages/kbn-ai-playground/utils/api.ts]: Of course

**Review by Samiul-TheSoccerFan — APPROVED** [review]
Looks good, tested locally. Ready to merge once the server codes moves to separate utils folder as suggested by @joemcelroy

**Pull request merged**